source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,198 @@
+===========================================
+Connections Survive Primary Step Down Tests
+===========================================
+
+These tests can be used to verify a driver's compliance with server discovery
+and monitoring requirements with respect to handling "not master" and
+"server shutting down" error responses from the server.
+
+These tests apply only to replica set topologies.
+
+Tests
+-----
+
+The driver should implement the following tests:
+
+getMore Iteration
+`````````````````
+
+This test requires server version 4.2 or higher, and makes use of events
+defined in the `CMAP specification
+<https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst>`_.
+
+Perform the following operations:
+
+- Insert 5 documents into a collection.
+- Clear the connection pool to the current primary (see CMAP spec for
+  the definition of connection pool clear operation).
+- Start a find operation on the collection with a batch size of 2, and
+  retrieve the first batch of results.
+- Verify that a ConnectionCreated CMAP event has been published.
```

Comment:
Since the following assertion is that a cmap event is NOT sent, it makes sense to assert that one is sent by the driver is some other circumstance, otherwise a driver which doesn't implement cmap events at all will pass the test. To guarantee that the event is sent, pool is cleared first. Would you like me to add this to the specification or remove the pool clear and publication verification entirely?

Reply:
I think it's sufficient for you to add a comment explaining what you just said (inline or in a rationale section).

Reply:
Added thank you.